File: csharp/ToolGood.Algorithm/Internals/Functions/Value/Function_ValueBoolean.cs
using System;
using System.Collections.Generic;
using System.Text;
using ToolGood.Algorithm.Enums;

namespace ToolGood.Algorithm.Internals.Functions.Value
{
	internal sealed class Function_ValueBoolean : FunctionBase
	{
		private readonly bool _value;
		public Function_ValueBoolean(bool value)
		{
			_value = value;
		}

		public override string Name => _value ? "True" : "False";

		public override Operand Evaluate(AlgorithmEngine engine, Func<AlgorithmEngine, string, Operand> tempParameter = null)
		{
			return _value ? Operand.True : Operand.False;
		}

		public override OperandType GetResultType()
		{
			return OperandType.BOOLEAN;
		}

		public override void ToString(StringBuilder stringBuilder, bool addBrackets)
		{
			stringBuilder.Append(Name);
		}
	}
}

File: java/toolgood.algorithm/src/main/java/toolgood/algorithm/internals/functions/value/Function_ValueBoolean.java
package toolgood.algorithm.internals.functions.value;

import java.util.function.BiFunction;
import toolgood.algorithm.AlgorithmEngine;
import toolgood.algorithm.Operand;
import toolgood.algorithm.enums.OperandType;
import toolgood.algorithm.internals.functions.Function_0;

public final class Function_ValueBoolean extends Function_0 {
    private final boolean _value;

    public Function_ValueBoolean(boolean value) {
        _value = value;
    }

    @Override
    public String Name() {
        return _value ? "True" : "False";
    }

    @Override
    public Operand Evaluate(AlgorithmEngine engine, BiFunction<AlgorithmEngine, String, Operand> tempParameter) {
        return _value ? Operand.True : Operand.False;
    }

    @Override
    public OperandType GetResultType() {
        return OperandType.BOOLEAN;
    }

    @Override
    public void ToString(StringBuilder stringBuilder, boolean addBrackets) {
        stringBuilder.append(Name());
    }
}
